### ISSUE TITLE
DuckDB jdbc driver fails to read a Struct type when loading json

### ISSUE THREAD

**Issue by kiritbasu** [duckdb/duckdb#6559]
### What happens?

duckdb 0.7.1 jdbc driver fails when trying to read a json file with nested structs 

### To Reproduce

- In DBeaver - create a new in-memory connection 
<img width="256" alt="image" src="https://user-images.githubusercontent.com/7513989/222799058-ba7a41ba-730b-405a-ad0f-2e4af827f656.png">
- Use the most recent JDBC driver
<img width="262" alt="image" src="https://user-images.githubusercontent.com/7513989/222799181-3340a97f-2fe4-42c4-b477-a9f35ee12eb1.png">

- in the editor Install and Load the JSON module
`INSTALL 'json';
LOAD 'json';`

- confirm the right version
`select version();` -> v0.7.1

- select from the json file
`select * from 'myfile.json';`
The first column is a struct and it fails with
`Caused by: java.sql.SQLException: Unsupported result column type STRUCT("$oid" VARCHAR)
	at org.duckdb.DuckDBNative.duckdb_jdbc_fetch(Native Method)
	at org.duckdb.DuckDBResultSet.next(DuckDBResultSet.java:79)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCResultSetImpl.next(JDBCResultSetImpl.java:272)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCResultSetImpl.nextRow(JDBCResultSetImpl.java:180)
	... 13 more`

When I do these exact steps directly in the duckdb console, I am able to read the data in just fine.
e.g.
![image](https://user-images.githubusercontent.com/7513989/222800110-46464b4a-5e00-4ead-b9db-35bbdafd9876.png)

### OS:

MacOS - arm64

### DuckDB Version:

0.7.1

### DuckDB Client:

JDBC

### Full Name:

Kirit Basu

### Affiliation:

Metaphor Data

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by DougHarmon** [comment]
I am getting  a similar error message as well

_SQL Error: Unsupported result column type STRUCT("version" UBIGINT)_

I'm also using dbeaver with duckdb jdbc driver version 0.7.1

Here are the contents of my version of myfile.json so that the error can be reproduced. 


```
{
    "Id": "1c8ab4d1-b8c3-4ced-a676-cb7f29f04bd6",
    "Request": {
        "version": 0
    },
    "Response": {
        "id": "e7336370-cb26-4962-95a8-1ec7a6f10568",
        "name": "Homer Simpson",
    },
    "BatchId": "e7336370-cb26-4962-95a8-1ec7a6f10568"
}
```

**Comment by dkapitan** [comment]
Same here, testing with https://www.hl7.org/fhir/examples-ndjson.zip using Python:

```python
from pathlib import Path
import duckdb


DATA = Path("../data")
TEST_PARQUET = Path("./output/parquet/")
TEST_DB = Path("./output/db.duckdb")

# test conversion ndjson to parquet with FHIR example files
for file_ in Path("./examples-ndjson/").glob("*.ndjson"):
    duckdb.read_json(file_.as_posix()).to_parquet((TEST_PARQUET / f"{file_.stem}.parquet").as_posix())

# read parquets into database
conn = duckdb.connect((TEST_DB).as_posix())
for file_ in TEST_PARQUET.glob("*.parquet"):
    stmt = f"""
        DROP TABLE IF EXISTS {file_.stem};
        CREATE TABLE {file_.stem} AS SELECT * FROM '{file_.as_posix()}'
        """
    conn.execute(stmt)
```

When inspecting with DbBeaver, I get the same error.

**Comment by Mause** [comment]
This should have been fixed in #7000
